backend/src/controllers/groupsController.js

```diff
@@ -10,23 +12,33 @@ function groupDataValid(group) {
 }
 
 function createGroup(req, res) {
-  const branchId = req.params.branchId;
   const group = {
+    id: uuid.v4(),
     name: req.body.name,
     description: req.body.description,
+    branchId: req.params.branchId,
   };
 
   if (!groupDataValid(group)) {
-    res.sendStatus(400);
-    return undefined;
+    return res.sendStatus(400);
   }
 
-  return groupService.create(group, branchId)
-    .then(groupData => res.status(200).json(groupData))
-    .catch(error => {
-      logger.error(`Failed creating a new group: branchId: ${branchId}}`, error);
-      res.sendStatus(500);
-    });
+  return branchService.findById(group.branchId)
+    .then(branch => {
+      if (!branch.id) {
+        res.sendStatus(400);
+        throw new Error();
+      }
+    })
+    .then(() => streamClient.publish('group-created', group))
+    .then(
+      () => res.status(200).json(group),
+      error => {
+        logger.error(`Failed creating a new group: ${group}}: `, error);
+        res.sendStatus(500);
+      }
+    )
+    .catch(() => {});
```

Comment:
This promise chain is pretty ugly. The `throw` is in there to skip the publish step, and then we catch the rejection at the bottom and ignore it, because we already sent the error status to the client.

Once branch persistence is converted it will all look a lot nicer, because we'll be able to check branch existence synchronously, so the awkward branching in the middle of the promise chain goes away.